### PR TITLE
FIX: Clicking on login should show login modal

### PR DESCRIPTION
**Pull request by nattsw**
Fixes: https://meta.discourse.org/t/log-in-button-on-subscription-page-does-not-work/280264

The login button's action wasn't linked properly to the component's defined action. Since using the `showLogin` route action does the same, we can use that instead.

Video of fix:

https://github.com/discourse/discourse-subscriptions/assets/1555215/ac1609be-bfd6-49c3-94ee-d1e4b7c7e245

### CODE REVIEW

File: assets/javascripts/discourse/templates/components/login-required.hbs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 <DButton
   @label="log_in"
-  @action={{this.createAccount}}
```
Comment (nattsw) [inline assets/javascripts/discourse/templates/components/login-required.hbs:5]: Technically speaking, `@action={{action "createAccount"}}` would have fixed it.

**Comment by tgxworld** [comment]
Do we need some sort of test here to catch regressions? For example, the `showLogin` route action may potentially be renamed in the future.

**Comment by nattsw** [comment]
Heh touché.

File: spec/system/subscription_product_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+describe "Subscription products", type: :system do
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:product) { Fabricate(:product, external_id: "prod_OiKyO6ZMFCIhQa") }
+
+  before { SiteSetting.discourse_subscriptions_enabled = true }
+
+  it "shows the login modal" do
+    p = DiscourseSubscriptions::Product.first
```
Comment (tgxworld) [inline spec/system/subscription_product_spec.rb:10]: Hmm is this required?

**Review by tgxworld — APPROVED** [review]

**Pull request merged**